### ISSUE TITLE
macOS regression: Unable to locate buku database

### ISSUE THREAD

**Issue by larsthegeek** [samhh/bukubrow-host#26]
Thanks so much for creating this tool.  It works like a charm on my linux box but I'm having issues on my Mac.

The host compiles without error but when executed simply reports "Failed to locate Buku database.".  The buku command works just fine.

Sadly, I'm not familiar enough with rust to submit a patch.

Cheers

**Comment by samhh** [comment]
Would you mind compiling at commit `1834eec54c6ade838c7900fd47cb0f838671f9a2` and seeing if that works?

**Comment by larsthegeek** [comment]
That did work.  Up and running on OSX and Linux.  Thank you!

**Comment by samhh** [comment]
Could I ask for two things to help me debug this @larsthegeek:

1. Output of: `echo $XDG_DATA_HOME`
2. Where is your `bookmarks.db`? Is it in `~/.local/share/buku/`?

**Comment by larsthegeek** [comment]
On my Mac (Mojave, zsh)

$XDG_DATA_HOME returns nothing

bookmarks.db is indeed in ~/.local/share/buku.

---
James Peterson, Programmer / SysAdmin / Really Nice Guy™

World Spice Merchants / www.worldspice.com
GPG: A2A0 D226 9740 81CB 8BA1 ACB2 51BF CBAC 67BC BA08

**Comment by samhh** [comment]
Note to self: Probably the case that we need to match the logic [here](https://github.com/jarun/buku/blob/master/buku#L385) (`get_default_dbdir`).

**Comment by naafx8** [comment]
I get the same problem.
I also already set xdg_data_home

![image](https://user-images.githubusercontent.com/23572835/90325084-578fec00-dfa1-11ea-8500-c6c767a2bbf9.png)

**Comment by samhh** [comment]
Hey guys, I have some time to look at this now. Could you checkout the `fix-macos-db-locating` branch and see if that fixes it for you?

**Comment by mridul-dev** [comment]
Have created the build out of the branch `fix-macos-db-locating` but still the same issue exists 

```
echo $XDG_DATA_HOME

ll ~/.local/share/buku/
total 24
-rw-r--r--  1 user1  host    12K Sep 13 16:35 bookmarks.db

./bukubrow-darwin-x64 --install-firefox
Failed to locate Buku database.
```

The build created from the Commit ID : `1834eec54c6ade838c7900fd47cb0f838671f9a2 ` works seamlessly.

**Comment by samhh** [comment]
One more question: What's the output of `echo $HOME`?

**Comment by mridul-dev** [comment]
Sure, thing. It is 

```
echo $HOME
/Users/user1
```

**Comment by samhh** [comment]
Cheers. I've pushed up a new commit that I hope will fix this. Same branch, though I amended the previous commit so you might need to force pull. Correct commit ID is `485b5af`. :crossed_fingers:

**Comment by mridul-dev** [comment]
@samhh : Thanks for the efforts in fixing this issue in a very short time. During the force pull I get a merge conflict in the file `src/buku/utils.rs`, so I deleted the local branch and did fresh checkout to the branch **fix-macos-db-locating** 

This issue is fixed 👍 

```
# git branch -vv
* fix-macos-db-locating 485b5af [origin/fix-macos-db-locating] Fix regression in locating database on macOS
  master                ab78758 [origin/master] Add support for Firefox on Windows (#24)

# make build-darwin-x64
# cd release && unzip build-darwin-x64.zip
# ./bukubrow-darwin-x64 --install-firefox
Successfully installed host for Firefox to:
	"/Users/user1/Library/Application Support/Mozilla/NativeMessagingHosts/com.samhh.bukubrow.json"
```

**Comment by samhh** [comment]
Huzzah! Thanks for your assistance, I'll merge that branch into master now.
